site/src/xServices/workspace/workspaceXService.ts

```diff
@@ -196,38 +198,33 @@ export const workspaceMachine = createMachine(
           ],
         },
       },
+      sseFailure: {
+        entry: ["assignRefreshWorkspaceError", "assignGetResourcesError"],
```

Comment:
My initial reason for separating getting and refreshing workspaces was in case we wanted the page to not load if there was an error getting the workspace the first time, but be stale-but-usable if there was an error refreshing. This might not be your doing but it looks like we're treating all resources errors the same but differentiating workspace errors. Is there a good reason for that?

Reply:
Yeah, good question. Before I touched this machine, we only loaded resources as part of the polling refresh. There was never any initial call to load resources on their own (unlike workspaces). So therefore, there's only one possible resource error, and that's if loading resources fails as part of the refresh effort.

So when I switched things over, I kept the same pattern. I think it makes sense because, to your point, we don't want to block the page from loading just because resources aren't coming back.

Maybe we can get rid of the refresh resources error completely and just rely on the refresh workspace error, considering they're now part of the same response object? If we did that, I would want to make sure we make an initial call to getResources outside of SSE, that would have it's own error handling, similar to workspace.

Let me know what you think would be best!

Reply:
Oh interesting. Does that mean resources are always delayed until the first refresh?

Reply:
yup!

Reply:
Do we definitely want to fetch the workspace outside of SSE first?

Reply:
Do we definitely want to fetch the workspace outside of SSE first?

Reply:
Idk if this helps, but the endpoints you normally get workspace data from should now have all data down to the workspace resources. I made all the workspace related data "fully render" so this may be easier to populate the initial data without another fetch. Take a look are the response data from getting the workspace and see if it has what we need.

Reply:
@presleyp @f0ssel As a results of this thread, I made the following decisions:

1. I am still fetching the workspace outside of SSE, because I need the workspace ID to initialize the EventSource, and I can't get that from the query params. @f0ssel what do you think about future us changing around the SSE endpoint such that it takes a workspace name and owner as params (as opposed to workspace id)? This would help us simplify the FE state. If you like that idea, I can make a ticket.
2. I got rid of the concept of a `resource` in this machine. Resources are considered part of workspaces and are not assigned separately or given separate error states. This made sense to me since we have normalized the workspace return object so that it will be identical across API calls - all workspace objects now have resources.
3. We now have two workspace failure states: `getWorkspaceError` which occurs if the initial `GET` request to fetch the workspace fails, and `refreshWorkspaceWarning`, which occurs if the SSE explodes for some reason (or returns an error event). `getWorkspaceError` blocks the page from loading but `refreshWorkspaceWarning` does not.

Reply:
Sounds great!